codice/ddf
# DDF-4750 Fix WFS 1.1.0 source paging

#### What does this PR do?
When a WFS 1.1.0 source is queried, it will now send a 'hits' query to the WFS server before it requests the results. (In WFS 1.1.0 there is no way to request the total hits and some number of results at the same time.) The source will now return the actual hit total in the response. Paging will now behave as expected on a WFS 1.1.0 source since the source knows the actual hit total.

#### Who is reviewing it? 
@glenhein 
@dcruver 
@leo-sakh 
@gjvera 
@bennuttle 

#### Select relevant component teams: 
@codice/ogc 

#### Ask 2 committers to review/merge the PR and tag them here.
@millerw8
@troymohl

#### How should this be tested?
1. In the Admin UI, open the Spatial app.
2. Click `Metacard to WFS Feature Map`.
3. For `Feature Type`, enter `{http://stategeothermaldata.org/uri-gin/aasg/xmlschema/welllog/0.8}AZWellLogs`.
4. For `Metacard Title to WFS Feature Property Mapping`, enter `DisplayName`.
5. Click `Save changes`.
6. Create a WFS 1.1.0 source with the URL http://services.azgs.az.gov/ArcGIS/services/aasggeothermal/AZWellLogs/MapServer/WFSServer.
7. Open Intrigue and run a * query against the source. Intrigue should say there are 2491 results.
8. Open the Table view and look at the `id` column. You should see the values `F3__1` through `F3__250`.
9. Click `Next` to go to the next page of results. Intrigue should still say there are 2491 results. You should see the values `F3__251` through `F3__500` in the `id` column.
10. Use the advanced search to search for `title CONTAINS *Lithologic*`. Intrigue should say there are 389 results.
11. Click `Next` to go to the next page of results. Intrigue should still say there are 389 results. There should be only 139 results on this page and you should not be able to click the `Next` button again.

#### What are the relevant tickets?
Fixes: #4750 

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
